exetera/core/fields.py

```diff
@@ -2922,11 +3096,36 @@ def data(self):
         self._ensure_valid()
         if self._value_wrapper is None:
             if self._write_enabled:
-                self._value_wrapper = WriteableFieldArray(self._field, 'values')
+                self._value_wrapper = WriteableFieldArray(self, 'values')
             else:
-                self._value_wrapper = ReadOnlyFieldArray(self._field, 'values')
+                self._value_wrapper = ReadOnlyFieldArray(self, 'values')
         return self._value_wrapper
 
+    def attach(self, view):
+        self._view_refs.append(view)
+
+    def detach(self, view=None):
+        if view is None:  # detach all
+            self._view_refs.clear()
+        else:
+            self._view_refs.remove(view)
+
+    def notify(self, msg=None):
+        for view in self._view_refs:
+            view.update(self, msg)
+
+    def update(self, subject, msg=None):
+        if isinstance(subject, (WriteableFieldArray, WriteableIndexedFieldArray)):
```

Comment:
`# This field is being notified by its own field array`
`# It needs to notify other fields that it is about to change before the change goes ahead`